services/pagerduty2/service.go

```diff
@@ -0,0 +1,328 @@
+package pagerduty2
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"sync/atomic"
+	"time"
+
+	"github.com/influxdata/kapacitor/alert"
+	"github.com/influxdata/kapacitor/keyvalue"
+	"github.com/influxdata/kapacitor/models"
+)
+
+// This example shows how to send a trigger event without a dedup_key.
+// In this case, PagerDuty will automatically assign a random and unique key
+// and return it in the response object.
+//
+// You should store this key in case you want to send an acknowledge or resolve
+// event to this incident in the future.
+//
+//{
+//    "payload": {
+//        "summary": "Example alert on host1.example.com",
+//        "timestamp": "2015-07-17T08:42:58.315+0000",
+//        "source": "monitoringtool:cloudvendor:central-region-dc-01:852559987:cluster/api-stats-prod-003",
+//        "severity": "info",
+//        "component": "postgres",
+//        "group": "prod-datapipe",
+//        "class": "deploy",
+//        "custom_details": {
+//            "ping time": "1500ms",
+//            "load avg": 0.75
+//        }
+//    },
+//    "routing_key": "samplekeyhere",
+//    "dedup_key": "samplekeyhere",
+//    "images": [{
+//        "src": "https://www.pagerduty.com/wp-content/uploads/2016/05/pagerduty-logo-green.png",
+//		  "href": "https://example.com/",
+//		  "alt": "Example text"
+//    }],
+//    "links": [{
+//        "href": "https://example.com/",
+//        "text": "Link text"
+//    }],
+//    "event_action": "trigger",
+//    "client": "Sample Monitoring Service",
+//    "client_url": "https://monitoring.example.com"
+//}
+
+// PDCEF is the PagerDuty - Common Event Format (PD-CEF) as outlined in the v2 API
+// https://v2.developer.pagerduty.com/docs/events-api-v2
+// https://support.pagerduty.com/docs/pd-cef
+//
+// API entry point is now https://events.pagerduty.com/v2/enqueue
+type PDCEF struct {
+	Summary       string
+	Source        string
+	Severity      string
+	Timestamp     time.Time
+	Class         string
+	Component     string
+	Group         string
+	CustomDetails string
```

Comment:
Based on the above it looks like this field can be arbitrary JSON, is that correct? If so the type could be `map[string]interface{}` or possibly just `interface{} `something and we could add the `details` JSON as this field.

See the VictorOps handler for how the details can be treated as rich data instead of just a string https://github.com/influxdata/kapacitor/blob/master/services/victorops/service.go#L208.  The `models.Result` type support JSON marshalling so it possibly could be set as the CustomDetails directly.

Reply:
Was following the existing `pagerduty` description which actually has this as a standard string. A local commit type set `CustomDetails` to the `map[string]interface{}` you called out because that does seem to be the most flexible.

https://github.com/influxdata/kapacitor/blob/master/services/pagerduty/service.go#L155

Do we want to change this one too?